File: quality/quality_sample.cpp
#include <iostream>
#include "opencv2/highgui.hpp"
#include "opencv2/quality.hpp"

int main(int argc, char **argv) {
    using namespace std;

    if(argc != 2) {
        cerr << "2 arguments required ./quality_sample <image path>" << endl;
        exit(1);
    }
    
    const string model_path = "./brisque_model_live.yml";
    const string range_path = "./brisque_range_live.yml";
    const cv::Mat input_image = cv::imread(argv[1], cv::IMREAD_COLOR);
    shared_ptr<cv::quality::QualityBase> quality(cv::quality::QualityBRISQUE::create(model_path, range_path));
    cv::Scalar score = quality->compute(input_image);
    cout << score[0] << endl;
    return 0;
}

File: quality/quality_sample.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import cv2
import sys

if __name__ == '__main__':
    # if len(sys.argv) != 2:
    #     print('usage: ./quality_sample.py <image path>')
    #     sys.exit(-1)

    model_path = "./brisque_model_live.yml"
    range_path = "./brisque_range_live.yml"
    input_image = cv2.imread(sys.argv[1])
    quality = cv2.quality.QualityBRISQUE_create(model_path, range_path)
    score = quality.compute([input_image])
    print(score)

    ref_image = cv2.imread(sys.argv[2])
    quality2 = cv2.quality.QualityMSE_create(ref_image)
    score2 = quality2.compute(input_image)
    print(score2[0])
